baseapp/baseapp.go

```diff
@@ -817,6 +843,19 @@ func (app *BaseApp) runTx(ctx sdk.Context, mode runTxMode, txBytes []byte) (gInf
 	// resources are acceessed by the ante handlers and message handlers.
 	defer acltypes.SendAllSignalsForTx(ctx.TxCompletionChannels())
 	acltypes.WaitForAllSignalsForTx(ctx.TxBlockingChannels())
+	// check for existing parent tracer, and if applicable, use it
+	spanCtx, tracer := ctx.SpanAndTracer()
+	if tracer != nil {
+		spanCtx, span := (*tracer).Start(spanCtx, "RunTx")
+		span.SetAttributes(attribute.String("txHash", fmt.Sprintf("%X", sha256.Sum256(txBytes))))
+		// update ctx span and tracer in case of further child tracing
+		ctx = ctx.WithSpanAndTracer(spanCtx, tracer)
+		defer span.End()
+	} else {
+		// this is unexpected, but we can just make a normal span
+		_, span := app.TracingInfo.Start("RunTx")
+		defer span.End()
```

Comment:
q: why is this unexpected? :o

Reply:
Because we will always be passing in a parent span for the initial delivertx call (see sei-chain abci.go)